src/OData.Cli/Program.cs

```diff
@@ -1,18 +1,87 @@
-﻿using Microsoft.Extensions.CommandLineUtils;
+﻿using System.CommandLine;
+using System.Threading.Tasks;
 
 namespace OData.Cli
 {
     class Program
     {
-        static void Main(string[] args)
+        static async Task Main(string[] args)
         {
-            var app = new CommandLineApplication();
-
-            app.Name = "odata-cli";
-            app.Description = "Client Proxy generator for OData endpoints.";
-
-            // Set the arguments to display the description and help text
-            app.HelpOption("-?|-h|--help");
+            var app = new RootCommand
+            {
+                new Command("generate", "Command to generate proxy classes for OData endpoints.")
+                {
+                    new Option<string>(
+                        new[] { "--metadata|-m"},
+                        "The URI of the metadata document. The value must be set to a valid service document URI or a local file path."
+                    ){IsRequired = true},
+                    new Option<string>(
+                        new[] { "--load-settings-from-file|-lsff" },
+                        "The URI to the settings file."
```

Comment:
```suggestion
                        "The location of the configuration settings file."
```
Because URI suggests a web location